naptime/src/main/scala/org/coursera/naptime/access/StructuredAccessControl.scala

```diff
@@ -44,8 +45,231 @@ case class StructuredAccessControl[A](
           Authorizer.toResponse(authorizer.authorize(decorated), decorated)
         }
       }.getOrElse {
-        Left(NaptimeActionException(Status.UNAUTHORIZED, Some("auth.perms"), Some("Missing authentication")))
+        StructuredAccessControl.missingResponse
       }
     }
   }
 }
+
+object StructuredAccessControl extends StructuredAccessControlCombinators {
+  private[access] val missingResponse =
+    Left(NaptimeActionException(Status.UNAUTHORIZED, Some("auth.perms"), Some("Missing authentication")))
+}
+
+trait StructuredAccessControlCombinators {
```

Comment:
I think these should be structured the same way as the authenticator combiners to keep everything consistent and maintainable. A few changes:
* Move these to a new `combiner` subpackage.
* Rename helpers to `anyOf`, and `eitherOf`.
* Move to separate files: AnyOf.scala, And.scala, EitherOf.scala.

(Also I'm not sure how I originally chose "combiner" vs. "combinator", but we should stick with one.)